### PR TITLE
DEVOPS-28230 if spec db version is empty take status db version instead if both empty throw error

### PR DESCRIPTION
**Pull request by EvertonCalgarotto**

### CODE REVIEW

File: pkg/databaseclaim/databaseclaim.go [hunk 1]
```diff
@@ -1150,7 +1150,7 @@ func (r *DatabaseClaimReconciler) createDatabaseAndExtensions(ctx context.Contex
 	return nil
 }
 
-func (r *DatabaseClaimReconciler) manageUserAndExtensions(ctx context.Context, reqInfo *requestInfo, logger logr.Logger, dbClient dbclient.Clienter, status *v1.Status, dbName string, baseUsername string) error {
+func (r *DatabaseClaimReconciler) manageUserAndExtensions(reqInfo *requestInfo, logger logr.Logger, dbClient dbclient.Clienter, status *v1.Status, dbName string, baseUsername string) error {
```
Comment (drewwells) [inline pkg/databaseclaim/databaseclaim.go:1153]: this method is performs remote network operation(s) that should accept cancel signals via context https://go.dev/blog/context
I'd keep context on it even though the underlying clients don't currently support canceling.

Keeping that in mind, what happens if db-controller restarts while running this method? It should support context Cancel so we can cleanly shutdown this operation.

File: pkg/hostparams/hostparams.go [hunk 2]
```diff
@@ -150,8 +154,13 @@ func New(config *viper.Viper, dbClaim *v1.DatabaseClaim) (*HostParams, error) {
 	}
 
 	if hostParams.EngineVersion == "" {
-		hostParams.isDefaultVersion = true
-		hostParams.EngineVersion = defaultEngineVersion
+		if dbClaim.Status.ActiveDB.DBVersion != "" {
+			log.Info("Using Status.ActiveDB.DBVersion as DefaultVersion")
```
Comment (drewwells) [inline pkg/hostparams/hostparams.go:158]: Not sure we want to log this, but Debug would be better. This is going to be logged a lot
```suggestion
			log.V(-1)Info("Using Status.ActiveDB.DBVersion as DefaultVersion")
```

File: pkg/hostparams/hostparams.go [hunk 3]
```diff
@@ -150,8 +154,13 @@ func New(config *viper.Viper, dbClaim *v1.DatabaseClaim) (*HostParams, error) {
...
+			hostParams.EngineVersion = dbClaim.Status.ActiveDB.DBVersion
+		} else {
+			hostParams.isDefaultVersion = true
```
Comment (drewwells) [inline pkg/hostparams/hostparams.go:161]: Do we still need this? I also thought you were going to set this in spec.dbVersion

File: pkg/hostparams/hostparams.go [hunk 4]
```diff
@@ -150,8 +154,13 @@ func New(config *viper.Viper, dbClaim *v1.DatabaseClaim) (*HostParams, error) {
 	}
 
 	if hostParams.EngineVersion == "" {
-		hostParams.isDefaultVersion = true
-		hostParams.EngineVersion = defaultEngineVersion
+		if dbClaim.Status.ActiveDB.DBVersion != "" {
+			log.Info("Using Status.ActiveDB.DBVersion as DefaultVersion")
+			hostParams.EngineVersion = dbClaim.Status.ActiveDB.DBVersion
```
Comment (drewwells) [inline pkg/hostparams/hostparams.go:159]: Modifies behavior of finding crossplane CR, so status is considered if it's available https://github.com/infobloxopen/db-controller/blob/625c742b7789cefb3c6e5b7934a162a9c4f13d6d/pkg/hostparams/hostparams.go#L63

File: pkg/hostparams/hostparams.go (hunk 3 above)
Comment (EvertonCalgarotto) [reply pkg/hostparams/hostparams.go:156]: no need, Spec.DBVersion might cause circular updates with Flux and DBController itself....where Flux will set it to empty and dbcontroller will set it to status.dbversion

File: pkg/databaseclaim/databaseclaim.go (hunk 1 above)
Comment (EvertonCalgarotto) [reply pkg/databaseclaim/databaseclaim.go:1153]: I'd put it back when needed.

**Review by drewwells — APPROVED** [review]

**Pull request merged**